### PR TITLE
support using mod_xsendfile and x-accel to download files

### PR DESCRIPTION
**Pull request by snopoke**

### CODE REVIEW

File: settings.py [hunk 1]
```diff
@@ -498,7 +498,28 @@
...
+SHARED_DRIVE_ROOT = None
+
+# name of the directory within SHARED_DRIVE_ROOT
+RESTORE_PAYLOAD_DIR_NAME = None
+
+if SHARED_DRIVE_ROOT and RESTORE_PAYLOAD_DIR_NAME:
+    # Defaults to tempfile.gettempdir()
+    RESTORE_PAYLOAD_DIR = os.path.join(SHARED_DRIVE_ROOT, RESTORE_PAYLOAD_DIR_NAME)
+
+####### django-transfer settings #######
```
Comment (dimagimon) [inline settings.py]: E265 block comment should start with '# '

**Comment by benrudolph** [comment]
just as a flag, you might need to update this file to set `SHARED_DRIVE_ROOT`: https://github.com/dimagi/commcarehq-ansible/blob/master/ansible/roles/commcarehq/templates/localsettings.py.j2

**Comment by dannyroberts** [comment]
Ah, that's so nice! Thanks @snopoke I get it now. I approve of everything you two are doing.

**Comment by snopoke** [comment]
Yes, but it should be OK to deploy this without that change since the
current localsettings sets "RESTORE_PAYLOAD_DIR" directly.

**Comment by benrudolph** [comment]
k looks good to me. not merging because of label

**Comment by czue** [comment]
yeah this is totally awesome. really hope it works!

File: settings.py [hunk 2]
```diff
@@ -498,7 +498,18 @@
     ]
 }
 
-RESTORE_PAYLOAD_DIR = None  # Defaults to tempfile.gettempdir()
+### Shared drive settings ###
```
Comment (dimagimon) [inline settings.py]: E265 block comment should start with '# '

File: settings.py [hunk 3]
```diff
@@ -498,7 +498,18 @@
...
+# Also see section after localsettings import
+SHARED_DRIVE_ROOT = None
+
+# name of the directory within SHARED_DRIVE_ROOT
+RESTORE_PAYLOAD_DIR_NAME = None
+
+## django-transfer settings
```
Comment (dimagimon) [inline settings.py]: E265 block comment should start with '# '

File: settings.py [hunk 4]
```diff
@@ -954,6 +965,17 @@
         **db_settings
     )
 
+### Shared drive settings ###
```
Comment (dimagimon) [inline settings.py]: E265 block comment should start with '# '

**Comment by snopoke** [comment]
@benrudolph :white_check_mark:

**Pull request merged**